docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/utils/DockerImageName.java

```diff
@@ -0,0 +1,491 @@
+            }
+        }
+    }
+    
+    public static class NamespaceName extends AbstractDockerImageName {
+        
+        /**
+         * Calls constructor on abstract to set namePart
+         * @param namePart Part of image name
+         */
+        public NamespaceName(String namePart) {
+            super(namePart);
+            this.partRegexp = "^([a-z0-9_-]*)$";
+            this.partInvertedRegexp = "[^a-z0-9_-]";
+        }
+        
+        @Override
+        public void validate() {
+            if (partName.length() < 2 ||partName.length() > 255) {
+                throw new DockerImageNameException("Invalid namespace: Namespace must be between 2 and 255 characters");
+            }
+            if (partName.startsWith("-") || partName.endsWith("-")) {
+                throw new DockerImageNameException("Invalid namespace: Namespace must not begin or end with a hyphon");
+            }
+            if (partName.contains("--")) {
+                throw new DockerImageNameException("Invalid namespace: Namespace must not contain consecutive hyphons");
+            }
+            if (!partName.matches(partRegexp)) {
+                throw new DockerImageNameException("Invalid namespace: Namespace must match [a-z0-9_-]");
+            }
+        }
+        
+        @Override
+        public boolean isValid() {
+            return partName.matches(partRegexp) && !partName.contains("--") 
+                    && !partName.startsWith("-") && !partName.endsWith("-") && partName.length() < 255
+                    && partName.length() > 2;
+        }
+        
+        @Override
+        public NamespaceName makeValid() {
+            if (partName.length() > 255) {
+                partName = partName.substring(0, 255);
+            }
+            if (!partName.matches(partRegexp)) {
+                partName = spacesToHyphens(partName)
+                        .toLowerCase()
+                        .replaceAll(partInvertedRegexp, "");
+            }
+            while (partName.contains("--")) {
+                partName = partName.replaceAll("--", "-");
+            }
+            partName = partName.replaceAll("^-", "").replaceAll("-$", "");
+            while (partName.length() < 2) {
+                partName += "0";
+            }
+            return this;
+        }
+    }
+    
+    public static class RepositoryName extends AbstractDockerImageName {
+        
+        /**
+         * @param namePart Part of image name
+         */
+        public RepositoryName(String namePart) {
+            super(namePart);
+            partRegexp = "^([a-z0-9_.-]*)$";
+            partInvertedRegexp = "[^a-z0-9_.-]";
+        }
+        
+        @Override
+        public void validate() {
+            if (!partName.matches(partRegexp)) {
+                throw new DockerImageNameException("Invalid registry: Registry must match [a-z0-9_.-]");
```

Comment:
isValid() returns boolean, validate() throws exceptions with the reason for the name being invalid - e.g. for form validation when users enter the repository name